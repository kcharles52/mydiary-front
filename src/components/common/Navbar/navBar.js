import React, { Component, Fragment } from "react";
import {
  Navbar,
  NavbarBrand,
  Collapse,
  NavItem,
  NavLink,
  NavbarToggler,
  Nav
} from "reactstrap";
import { authService } from "../../../utils/auth";
export class NavigationBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  render() {
    const isLoggedIn = authService.isAuthenticated;
    return (
      <Fragment>
        <Navbar color="primary" light expand="md">
          <NavbarBrand href="/home">My Diary</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse navbar isOpen={this.state.isOpen}>
            <Nav className="ml-auto" navbar hidden={!isLoggedIn}>
              <NavItem>
                <NavLink href="/home">Home</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/addEntry">Create Journel</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/" onClick={authService.logoutUser}>
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </Fragment>
    );
  }
}
export default NavigationBar;
