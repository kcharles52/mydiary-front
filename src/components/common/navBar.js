import React, { Component, Fragment } from "react";
import {
  Navbar,
  NavbarBrand,
  Collapse,
  NavItem,
  NavLink,
  NavbarToggler,
  Nav
} from "reactstrap";
import { Authenticate } from "../../routes/protectedRoutes";

class NavigationBar extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      isOpen: false
    };
  }
  handleClick = () => {
    localStorage.clear();
  };
  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    const isLoggedIn = Authenticate(localStorage.getItem("token"));
    return (
      <Fragment>
        <Navbar color="primary" light expand="md">
          <NavbarBrand href="/">My Diary</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse navbar isOpen={this.state.isOpen}>
            <Nav className="ml-auto" navbar hidden={!isLoggedIn}>
              <NavItem>
                <NavLink href="/home">Home</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/addEntry">Create Journel</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/home">Diary Entries</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/" onClick={this.handleClick}>
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </Fragment>
    );
  }
}
export default NavigationBar;
