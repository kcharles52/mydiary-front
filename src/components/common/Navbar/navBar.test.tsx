// react libraries
import * as React from "react";

// third party libraries
import { mount, shallow } from "enzyme";

//components
import { NavigationBar } from "./navBar";

describe("Render navigation bar", () => {
  let wrapper, shallowWrapper;
  beforeEach(() => {
    wrapper = mount(<NavigationBar />);
    shallowWrapper = shallow(<NavigationBar />);
  });
  afterEach(() => {
    wrapper = shallowWrapper = null;
  });

  it("should render the navbar corrextly", () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
  it("should toggle navbar", () => {
    wrapper.setState({ isOpen: false });
    wrapper.instance().toggle();
    expect(wrapper.instance().state.isOpen).toBe(true);
  });
  it("should logout user", () => {
    wrapper
      .find("a")
      .last()
      .simulate("click");
    // expect(wrapper.instance().handleClick()).toBeCalled();
  });
});
