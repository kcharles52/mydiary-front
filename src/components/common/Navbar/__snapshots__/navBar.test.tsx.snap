// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render navigation bar should render the navbar corrextly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Navbar
        color="primary"
        expand="md"
        light={true}
        tag="nav"
      >
        <NavbarBrand
          href="/home"
          tag="a"
        >
          My Diary
        </NavbarBrand>
        <NavbarToggler
          onClick={[Function]}
          tag="button"
          type="button"
        />
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
        >
          <Nav
            className="ml-auto"
            hidden={false}
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                href="/home"
                tag="a"
              >
                Home
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                href="/addEntry"
                tag="a"
              >
                Create Journel
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                href="/"
                onClick={[Function]}
                tag="a"
              >
                Logout
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavbarBrand
            href="/home"
            tag="a"
          >
            My Diary
          </NavbarBrand>,
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              hidden={false}
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/home"
                  tag="a"
                >
                  Home
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/addEntry"
                  tag="a"
                >
                  Create Journel
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/"
                  onClick={[Function]}
                  tag="a"
                >
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>,
        ],
        "color": "primary",
        "expand": "md",
        "light": true,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "My Diary",
            "href": "/home",
            "tag": "a",
          },
          "ref": null,
          "rendered": "My Diary",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              hidden={false}
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/home"
                  tag="a"
                >
                  Home
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/addEntry"
                  tag="a"
                >
                  Create Journel
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/"
                  onClick={[Function]}
                  tag="a"
                >
                  Logout
                </NavLink>
              </NavItem>
            </Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/home"
                    tag="a"
                  >
                    Home
                  </NavLink>
                </NavItem>,
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/addEntry"
                    tag="a"
                  >
                    Create Journel
                  </NavLink>
                </NavItem>,
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/"
                    onClick={[Function]}
                    tag="a"
                  >
                    Logout
                  </NavLink>
                </NavItem>,
              ],
              "className": "ml-auto",
              "hidden": false,
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    href="/home"
                    tag="a"
                  >
                    Home
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "href": "/home",
                    "tag": "a",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    href="/addEntry"
                    tag="a"
                  >
                    Create Journel
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Create Journel",
                    "href": "/addEntry",
                    "tag": "a",
                  },
                  "ref": null,
                  "rendered": "Create Journel",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    href="/"
                    onClick={[Function]}
                    tag="a"
                  >
                    Logout
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "href": "/",
                    "onClick": [Function],
                    "tag": "a",
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Navbar
          color="primary"
          expand="md"
          light={true}
          tag="nav"
        >
          <NavbarBrand
            href="/home"
            tag="a"
          >
            My Diary
          </NavbarBrand>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              hidden={false}
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/home"
                  tag="a"
                >
                  Home
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/addEntry"
                  tag="a"
                >
                  Create Journel
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/"
                  onClick={[Function]}
                  tag="a"
                >
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavbarBrand
              href="/home"
              tag="a"
            >
              My Diary
            </NavbarBrand>,
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            />,
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                className="ml-auto"
                hidden={false}
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/home"
                    tag="a"
                  >
                    Home
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/addEntry"
                    tag="a"
                  >
                    Create Journel
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/"
                    onClick={[Function]}
                    tag="a"
                  >
                    Logout
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>,
          ],
          "color": "primary",
          "expand": "md",
          "light": true,
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "My Diary",
              "href": "/home",
              "tag": "a",
            },
            "ref": null,
            "rendered": "My Diary",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "tag": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": <Nav
                className="ml-auto"
                hidden={false}
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/home"
                    tag="a"
                  >
                    Home
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/addEntry"
                    tag="a"
                  >
                    Create Journel
                  </NavLink>
                </NavItem>
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="/"
                    onClick={[Function]}
                    tag="a"
                  >
                    Logout
                  </NavLink>
                </NavItem>
              </Nav>,
              "enter": true,
              "exit": true,
              "in": false,
              "isOpen": false,
              "mountOnEnter": false,
              "navbar": true,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 350,
              "unmountOnExit": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="/home"
                      tag="a"
                    >
                      Home
                    </NavLink>
                  </NavItem>,
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="/addEntry"
                      tag="a"
                    >
                      Create Journel
                    </NavLink>
                  </NavItem>,
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="/"
                      onClick={[Function]}
                      tag="a"
                    >
                      Logout
                    </NavLink>
                  </NavItem>,
                ],
                "className": "ml-auto",
                "hidden": false,
                "navbar": true,
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      href="/home"
                      tag="a"
                    >
                      Home
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "href": "/home",
                      "tag": "a",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      href="/addEntry"
                      tag="a"
                    >
                      Create Journel
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Create Journel",
                      "href": "/addEntry",
                      "tag": "a",
                    },
                    "ref": null,
                    "rendered": "Create Journel",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      href="/"
                      onClick={[Function]}
                      tag="a"
                    >
                      Logout
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Logout",
                      "href": "/",
                      "onClick": [Function],
                      "tag": "a",
                    },
                    "ref": null,
                    "rendered": "Logout",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
